### PR TITLE
Add more tests from HackMD examples 

### PR DESCRIPTION
**Pull request by domenicquirl**
Adds the examples `killing-and-murder` and `canonical-liveness`.

The facts of the former are interesting, because the include facts from a dereference assignment `*y = 11`, where it may be interesting to talk about what facts to generate (see discussion [on zulip](https://rust-lang.zulipchat.com/#narrow/stream/186049-t-compiler.2Fwg-polonius/topic/purely.20forward.20propagation/near/256393101)).

### CODE REVIEW

**Review by nikomatsakis — COMMENTED** [review]
I *think* these look right! I have to admit that reading the lists of facts kinds of make my eyes glaze over after a time, seems important to get that front-end working.

File: tests/canonical-liveness-err/program.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    
+    // Invalidate borrows of `p`
+    invalidate_origin('L_p)
+
+    goto d
+}
+
+d: "use(x)" {
+    // Access `x` [== `&q`] where `x: &'x mut u32`
+    access_origin('x)
```
[the remark this answers is not in the document]
Comment (domenicquirl) [reply tests/canonical-liveness-err/program.txt:43]: This version does indeed produce an error (as it should) - see the comment in line 6 and the above output (the one below is for the non-erroneous version).

File: tests/canonical-liveness-err/program.txt [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// Corresponds to `canonical-liveness`, but with the "saving" re-assignment to `x` removed.
+
+// let p = 22;
```
Comment (lqd) [inline tests/canonical-liveness-err/program.txt:3]:
```suggestion
// let mut p = 22;
```

File: tests/canonical-liveness-err/program.txt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+// let mut x: &i32 = &p; // `x` points at `p`
```
Comment (lqd) [inline tests/canonical-liveness-err/program.txt:4]:
```suggestion
// let x: &u32 = &p; // `x` points at `p`
```

File: tests/canonical-liveness-err/program.txt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    // Access `p` where `p: u32`
+    
+    // Invalidate borrows of `p`
+    invalidate_origin('L_p)
+
+    goto d
+}
+
+d: "use(x)" {
+    // Access `x` [== `&q`] where `x: &'x mut u32`
```
Comment (lqd) [inline tests/canonical-liveness-err/program.txt:42]:
```suggestion
    // Access `x` [== `&q`] where `x: &'x u32`
```

File: tests/killing-and-murder-err/program.txt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    invalidate_origin('L_*x)
+
+    // Clear all origins in `y` (overwritten plan):
+    clear_origin('y)
+
+    // Clear 'L_*x because it's being recreated
+    clear_origin('L_*x)
+
+    // Unroll `*x`
+    introduce_subset('L_*x, 'x)
```
[the remark this answers is not in the document]
Comment (domenicquirl) [reply tests/killing-and-murder-err/program.txt:47]: Yeah, I though that might have been it when I wrote the doc. The "unroll" term is from Niko's description in the rules comment. My understanding of this part of the facts is [this part](https://hackmd.io/KpK0ICQOQLSPxIcNIlhkJQ?view#Borrows) of the HackMD additions if you want to have a look already.

**Pull request merged**